### ISSUE TITLE
Flutter plugin changes the SDK of non flutter projects

### ISSUE THREAD

**Issue by leafpetersen** [flutter/flutter-intellij#565]
Using the Flutter plugin changes the SDK used for other non-flutter projects.  For example:

Open a non flutter project (say the analyzer), and set the Dart SDK to something specific:

<img width="1057" alt="screen shot 2016-12-16 at 1 38 34 pm" src="https://cloud.githubusercontent.com/assets/8484504/21279418/f910001a-c394-11e6-8746-5f7f415cc1f1.png">

Now configure the Flutter plugin to point to a flutter SDK:

<img width="1027" alt="screen shot 2016-12-16 at 1 39 59 pm" src="https://cloud.githubusercontent.com/assets/8484504/21279456/2a112900-c395-11e6-93fa-0393fc93ec82.png">

The Dart configuration for the non-flutter project has now changed:

<img width="1032" alt="screen shot 2016-12-16 at 1 41 12 pm" src="https://cloud.githubusercontent.com/assets/8484504/21279501/568b31e2-c395-11e6-865e-59fcd8631a0c.png">

The same issue can be triggered via the New Project workflow as well.

Open a non-Flutter project (say the analyzer), configure the Dart SDK appropriately, then create a new Flutter project in a new window.  Switch back to the old non-flutter project, and the Dart SDK for that project is now the Flutter SDK.

Intellij 2016.3.1 on MacOS

```
Flutter • channel master • https://github.com/flutter/flutter.git Framework • revision 2546259a64 (23 hours ago) • 2016-12-15 15:05:15 Engine • revision 2f32e8054d Tools • Dart 1.21.0
```

**Comment by leafpetersen** [comment]
This also goes the other way too.  If you switch from a flutter project window to a non-flutter project window, notice that your SDK is wrong and fix it, then when you switch back to the flutter project the Dart SDK for that flutter project seems to be switched over to the non-flutter SDK.

**Comment by pq** [comment]
Thanks @leafpetersen !

I agree this is unfortunate.  It's a side-effect of the fact that the Dart plugin treats the SDK as an application-level (singleton) setting.  We've talked a bit about this and I believe that @alexander-doroshko is looking into evolving it to support per-project (if not per-module?) SDKs.

A wee bit more context here: #56.

@jwren may have thoughts too.

**Comment by alexander-doroshko** [comment]
Tracked as https://youtrack.jetbrains.com/issue/WEB-20870, planned for 2017.1

**Comment by devoncarew** [comment]
Fixed upstream in the Dart plugin (and available now in 2017.1).

**Comment by leafpetersen** [comment]
Woohoo! Thanks!
